eth/tracers/native/erc7562.go

```diff
@@ -0,0 +1,595 @@
+		ContractSize:      map[common.Address]*contractSizeWithOpcode{},
+	}
+	if depth == 0 {
+		call.Gas = t.gasLimit
+	}
+	t.callstackWithOpcodes = append(t.callstackWithOpcodes, call)
+}
+
+func (t *erc7562Tracer) captureEnd(output []byte, gasUsed uint64, err error, reverted bool) {
+	if len(t.callstackWithOpcodes) != 1 {
+		return
+	}
+	t.callstackWithOpcodes[0].processOutput(output, err, reverted)
+}
+
+// OnExit is called when EVM exits a scope, even if the scope didn't
+// execute any code.
+func (t *erc7562Tracer) OnExit(depth int, output []byte, gasUsed uint64, err error, reverted bool) {
+	defer catchPanic()
+	if depth == 0 {
+		t.captureEnd(output, gasUsed, err, reverted)
+		return
+	}
+
+	t.depth = depth - 1
+	if t.config.OnlyTopCall {
+		return
+	}
+
+	size := len(t.callstackWithOpcodes)
+	if size <= 1 {
+		return
+	}
+	// Pop call.
+	call := t.callstackWithOpcodes[size-1]
+	t.callstackWithOpcodes = t.callstackWithOpcodes[:size-1]
+	size -= 1
+
+	call.GasUsed = gasUsed
+	call.processOutput(output, err, reverted)
+	// Nest call into parent.
+	t.callstackWithOpcodes[size-1].Calls = append(t.callstackWithOpcodes[size-1].Calls, call)
+}
+
+func (t *erc7562Tracer) OnTxEnd(receipt *types.Receipt, err error) {
+	defer catchPanic()
+	// Error happened during tx validation.
+	if err != nil {
+		return
+	}
+	t.callstackWithOpcodes[0].GasUsed = receipt.GasUsed
+	if t.config.WithLog {
+		// Logs are not emitted when the call fails
+		clearFailedLogs2(&t.callstackWithOpcodes[0], false)
+	}
+}
+
+func (t *erc7562Tracer) OnLog(log1 *types.Log) {
+	defer catchPanic()
+	// Only logs need to be captured via opcode processing
+	if !t.config.WithLog {
+		return
+	}
+	// Avoid processing nested calls when only caring about top call
+	if t.config.OnlyTopCall && t.depth > 0 {
+		return
+	}
+	// Skip if tracing was interrupted
+	if t.interrupt.Load() {
+		return
+	}
+	l := callLog{
+		Address:  log1.Address,
+		Topics:   log1.Topics,
+		Data:     log1.Data,
+		Position: hexutil.Uint(len(t.callstackWithOpcodes[len(t.callstackWithOpcodes)-1].Calls)),
+	}
+	t.callstackWithOpcodes[len(t.callstackWithOpcodes)-1].Logs = append(t.callstackWithOpcodes[len(t.callstackWithOpcodes)-1].Logs, l)
+}
+
+func (t *erc7562Tracer) handleLogs(opcode vm.OpCode, scope tracing.OpContext) {
+	defer catchPanic()
+	if opcode == vm.LOG0 || opcode == vm.LOG1 || opcode == vm.LOG2 || opcode == vm.LOG3 || opcode == vm.LOG4 {
+	}
+	//	count := int(opcode - vm.LOG0)
+	//	ofs := peepStack(scope.StackData(), 0)
+	//	len := peepStack(scope.StackData(), 1)
+	//	memory := scope.MemoryData()
+	//	topics := []hexutil.Bytes{}
+	//	for i := 0; i < count; i++ {
+	//		topics = append(topics, peepStack(scope.StackData(), 2+i).Bytes())
+	//		//topics = append(topics, scope.Stack.Back(2+i).Bytes())
+	//	}
+	//	log := make([]byte, len.Uint64())
+	//	copy(log, memory[ofs.Uint64():ofs.Uint64()+len.Uint64()])
+	//	t.Logs = append(t.Logs, &logsItem{
+	//		Data:  log,
+	//		Topic: topics,
+	//	})
+	//}
```

Comment:
A reminder that this is still commented out.

Reply:
Yes I'm actually still on the fence if we need it or not

Reply:
Do we want to proactively handle logs for future possible need?